picard/script.py

```diff
@@ -910,6 +910,143 @@ def func_is_video(parser):
         return ""
 
 
+def func_find_str(parser, haystack, needle):
+    """Find the location of the first occurrence of one string within another.
+
+    Arguments:
+        haystack {str} -- The string to search
+        needle {str} -- The substring to find
+
+    Returns:
+        str -- Returns the zero-based index of the first occurrance of needle in haystack, or -1 if needle was not found.
+    """
+    return str(haystack.find(needle))
+
+
+def func_reverse_str(parser, text):
+    """Returns 'text' in reverse order.
+
+    Arguments:
+        text {str} -- Text to be processed
+
+    Returns:
+        str -- Text in reverse order.
+    """
+    return text[::-1]
+
+
+def func_substr(parser, text, start_index, end_index):
+    """Extract a specified portion of a string.
+
+    Arguments:
+        text {str} -- The string from which the extract will be made.
+        start_index {int} -- Index of the first character to extract.
+        end_index {int} -- Index of the first character that will not be extracted.
+
+    Returns:
+        str -- Returns the substring beginning with the character at the start index,
+               up to (but not including) the character at the end index.  The first
+               character is at index number 0.  If the start index is left blank, it
+               defaults to the first character in the string.  If the end index is
+               left blank, it defaults to the number of characters in the string.
+               If either index is negative, it is subtracted from the total number of
+               characters in the string to provide the index used.
+    """
+    _start = int(start_index) if start_index else None
+    _end = int(end_index) if end_index else None
```

Comment:
This must catch `ValueError` in case the int conversion fails. I'd say we treat the value throwing the exception should just be considered `None`.

Reply:
Updated to include the ValueError check.